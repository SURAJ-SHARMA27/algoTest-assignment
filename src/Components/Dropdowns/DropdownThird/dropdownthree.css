/* Container for the div */
@import url('https://fonts.googleapis.com/css2?family=Inter:wght@400;600&display=swap');

.dropdown-container-three {
  width: 12rem;
  font-family: Arial, sans-serif;
}

/* Static header */
.dropdown-header {
  background-color: #FAFAFA;
  padding: 10px;
  height: 44px;
  border: 1px solid #E5E7EB;
  
  display: flex;
  align-items: center;
}

/* Title and value styles */
.dropdown-title {
  font-family: 'Inter', sans-serif;
  font-size: 12px;
  color: #3A3A3A;
  margin: 0;
}

.dropdown-value {
  font-family: 'Inter', sans-serif;
  font-size: 12px;
  color: #3A3A3A;
  font-weight: 600;
  margin: 0;
}
