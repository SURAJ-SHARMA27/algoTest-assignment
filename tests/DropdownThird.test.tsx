import React from 'react';
import { render } from '@testing-library/react';
import DropdownThird from '../src/Components/Dropdowns/DropdownThird/DropdownThree'; // Adjust the import path as needed
import { it, expect, describe } from 'vitest';
import '@testing-library/jest-dom/vitest';

describe('DropdownThird Component', () => {
    it('should render without crashing', () => {
        render(<DropdownThird />);
    });

    it('should display the correct selected option name and value', () => {
        const { getByText } = render(<DropdownThird />);
         expect(getByText('India VIX')).toBeInTheDocument();
         expect(getByText('12.92')).toBeInTheDocument();
    });

    it('matches the snapshot', () => {
        const { asFragment } = render(<DropdownThird />);
        expect(asFragment()).toMatchSnapshot();
    });
});
