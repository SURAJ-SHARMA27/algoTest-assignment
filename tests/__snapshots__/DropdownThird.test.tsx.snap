// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DropdownThird Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="dropdown-container-three"
  >
    <div
      class="dropdown-header"
    >
      <div
        style="flex-direction: column; display: flex; align-items: flex-start;"
      >
        <div
          class="dropdown-title"
        >
          India VIX
        </div>
        <div
          class="dropdown-value"
        >
          12.92
          <span
            class="green"
          >
             0.00
          </span>
           
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
